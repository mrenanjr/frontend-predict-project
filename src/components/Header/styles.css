.header-container {
    height: 80px;
    width: 100%;

    display: flex;
    justify-content: space-between;
    align-items: center;

    background: linear-gradient(180deg, rgba(255, 255, 255, 0) 0%, rgba(0, 103, 172, 0.1) 100%), linear-gradient(180deg, #0067AC 0%, rgba(255, 255, 255, 0) 100%);
}

.header-container .img {
    height: 80px;
    width: 150px;

    background: #C0C0C0;
    
    margin-left: 60px;
}

.header-container .img img {
    margin: auto;
    display: block;
    height: 100%;
}

.header-container .menu a {
    padding: 20px;
    color: white;
    cursor: pointer;

    text-shadow: 0px 4px 10px rgba(0, 0, 0, 0.6);
    
    transition: 0.2s;
}

.header-container .menu a:hover {
    opacity: 0.5;
}

.header-container p {
    margin: 0;

    font-size: 30px;
    font-weight: 500;
    color: #0F084B;
    text-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
}