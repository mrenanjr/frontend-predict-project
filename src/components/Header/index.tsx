import React from 'react';
import { Link } from 'react-router-dom';
import { Select } from 'semantic-ui-react';

import './styles.css';

import Img from '../../assets/insert_photo-24px.svg';
import HomeIcon from '../../assets/home-24px.svg';
import InfoIcon from '../../assets/info-24px.svg';
import QuitIcon from '../../assets/power_settings_new-24px.svg';

const Header = () => {
    const instOptions = [
        { key: 'uni', value: 'UNI', text: 'UNI' }
    ];
    return (
        <header className="header-container">
            <div className="img">
                <Link to="/inicio">
                    <img src={Img} alt="Imagem"/>
                </Link>
            </div>
            <Link to="/inicio">
                <p>S.I.S.A.P</p>
            </Link>
            <div className="menu">
                <Select placeholder='Selecione a Instituição' value='UNI' options={instOptions} />
                <Link to="/inicio">
                    <img src={HomeIcon} alt="Home" />
                </Link>
                <Link to="/sobrenos">
                    <img src={InfoIcon} alt="Informação" />
                </Link>
                <Link to="/" style={{ marginRight: 60 }}>
                    <img src={QuitIcon} alt="Sair" />
                </Link>
            </div>
        </header>
    );
}

export default Header;
