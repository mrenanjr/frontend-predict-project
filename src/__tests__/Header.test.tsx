import * as React from 'react';
import { shallow } from 'enzyme';
import Header from '../components/Header';

test('Header loads corretly', () => {
  const wrapper = shallow(<Header />);

  expect(wrapper).toMatchSnapshot();
});