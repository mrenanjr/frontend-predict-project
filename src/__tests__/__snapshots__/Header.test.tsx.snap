// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header loads corretly 1`] = `
<header
  className="header-container"
>
  <div
    className="img"
  >
    <Link
      to="/inicio"
    >
      <img
        alt="Imagem"
        src="test-file-stub"
      />
    </Link>
  </div>
  <Link
    to="/inicio"
  >
    <p>
      SISAP
    </p>
  </Link>
  <div
    className="menu "
  >
    <div
      className="menu-toogle"
      onClick={[Function]}
    >
      <div
        className="one"
      />
      <div
        className="two"
      />
      <div
        className="three"
      />
    </div>
    <div
      className="elements"
    >
      <Select
        options={
          Array [
            Object {
              "key": "ies",
              "text": "IES",
              "value": "IES",
            },
          ]
        }
        placeholder="Selecione a Instituição"
        value="IES"
      />
      <Link
        onClick={[Function]}
        to="/inicio"
      >
        <img
          alt="Home"
          src="test-file-stub"
        />
      </Link>
      <Link
        onClick={[Function]}
        to="/sobrenos"
      >
        <img
          alt="Informação"
          src="test-file-stub"
        />
      </Link>
      <a
        onClick={[Function]}
      >
        <img
          alt="Sair"
          src="test-file-stub"
        />
      </a>
    </div>
  </div>
</header>
`;
